gneiss/tests/test_util.py

```
@@ -345,6 +347,37 @@ def test_random_tree(self):
                'y7:0.0310177678435)y4:0.106894706976)y2:0.15840160938)y0;\n')
         self.assertEqual(str(t), exp)
 
+    def test_block_diagonal_4x4(self):
+        np.random.seed(0)
+        res = block_diagonal(4, 4, 2)
+        exp = np.array([[0.5488135, 0.71518937, 0., 0.],
+                        [0.60276338, 0.54488318, 0., 0.],
+                        [0., 0., 0.4236548, 0.64589411],
+                        [0., 0., 0.43758721, 0.891773]])
+        npt.assert_allclose(res, exp, rtol=1e-5, atol=1e-5)
+
+    def test_block_diagonal_3x4(self):
```

It should throw an error, because that is outside the scope of this function